tools/chplcheck/src/fixits.py

```diff
@@ -0,0 +1,90 @@
+#
+# Copyright 2024 Hewlett Packard Enterprise Development LP
+# Other additional copyright holders may be indicated within.
+#
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import chapel
+from dataclasses import dataclass
+import typing
+
+
+@dataclass
+class ChapelFixit:
+    """
+    Represents a fixit for a Chapel diagnostic.
+    """
+
+    path: str
+    start: typing.Tuple[int, int]
+    end: typing.Tuple[int, int]
+    text: str
+
+    @classmethod
+    def build(cls, location: chapel.Location, text: str) -> "ChapelFixit":
+        return ChapelFixit(
+            location.path(), location.start(), location.end(), text
+        )
+
+    @classmethod
+    def to_dict(cls, fixit: "ChapelFixit") -> typing.Dict:
+        return {
+            "location": {
+                "path": fixit.path,
+                "start": fixit.start,
+                "end": fixit.end,
+            },
+            "text": fixit.text,
+        }
+
+    @classmethod
+    def from_dict(cls, data: typing.Dict) -> typing.Optional["ChapelFixit"]:
+
+        if "location" not in data or "text" not in data:
+            return None
+
+        location = data["location"]
+        if (
+            "path" not in location
+            or "start" not in location
+            or "end" not in location
+        ):
+            return None
+
+        return ChapelFixit(
+            location["path"], location["start"], location["end"], data["text"]
+        )
+
+
+def range_to_text(rng: chapel.Location, lines: typing.List[str]) -> str:
+    """
+    Convert a Chapel location to a string. If the location spans multiple
+    lines, it gets truncated into 1 line. The lines and columns are
+    zero-indexed.
+    """
+
+    (line_start, char_start) = rng.start()
+    (line_end, char_end) = rng.end()
+
+    if line_start == line_end:
+        return lines[line_start - 1][char_start - 1 : char_end - 1]
+
+    text = [lines[line_start - 1][char_start - 1 :]]
+    for line in range(line_start + 1, line_end):
+        text.append(lines[line - 1])
+    text.append(lines[line_end - 1][: char_end - 1])
+
+    return "\n".join([t for t in text])
```

Comment:
We don't have this anywhere else? I'm struggling to believe that.

Reply:
I see this in `chpl-language-server.py`. I think you should move `range_to_tokens` and `range_to_text` to `chapel`. `range_to_tokens` may seem suspect to move, but there's nothing about it that's specific to the language server protocol. All it does is "pain" input lines with the location (alternatively, it splits a given location into ranges determined by given lines).